File: kotlin/lib/src/main/kotlin/models/AzureBlobStorageConfigIn.kt
// This file is @generated
package com.svix.kotlin.models

import kotlinx.serialization.Serializable

@Serializable
data class AzureBlobStorageConfigIn(
    val container: String,
    val account: String,
    /**
     * Access key.
     *
     * Currently a required field, but marked as optional because we may add different
     * authentication in the future.
     */
    val accessKey: String? = null,
)

File: javascript/src/models/azureBlobStorageConfigIn.ts
// this file is @generated

export interface AzureBlobStorageConfigIn {
  container: string;
  account: string;
  /**
   * Access key.
   *
   * Currently a required field, but marked as optional because we may add different authentication in the future.
   */
  accessKey?: string | null;
}

export const AzureBlobStorageConfigInSerializer = {
  _fromJsonObject(object: any): AzureBlobStorageConfigIn {
    return {
      container: object["container"],
      account: object["account"],
      accessKey: object["accessKey"],
    };
  },

  _toJsonObject(self: AzureBlobStorageConfigIn): any {
    return {
      container: self.container,
      account: self.account,
      accessKey: self.accessKey,
    };
  },
};
